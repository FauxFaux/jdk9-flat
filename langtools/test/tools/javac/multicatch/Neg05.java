/*
 * @test /nodynamiccopyright/
 * @bug 6943289
 *
 * @summary Project Coin: Improved Exception Handling for Java (aka 'multicatch')
 * @author mcimadamore
 * @compile/fail/ref=Neg05.out -XDrawDiagnostics Neg05.java
 *
 */

class Neg02 {

    static class Foo<X> {
       Foo(X x) {}
    }

    static interface Base<X> {}
    static class A extends Exception implements Base<String> {}
    static class B extends Exception implements Base<Integer> {}

    void m() {
        try {
            if (true) {
                throw new A();
            }
            else {
                throw new B();
            }
        } catch (A | B ex) {
            Foo<?> f = new Foo<>(ex);
        }
    }
}
