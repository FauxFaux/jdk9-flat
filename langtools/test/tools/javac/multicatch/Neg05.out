Neg05.java:30:31: compiler.err.cant.apply.diamond.1: (compiler.misc.diamond: Neg02.Foo), (compiler.misc.diamond.invalid.arg: java.lang.Exception&Neg02.Base<? extends java.lang.Object&java.io.Serializable&java.lang.Comparable<? extends java.lang.Object&java.io.Serializable&java.lang.Comparable<?>>>, (compiler.misc.diamond: Neg02.Foo))
1 error
